### PR TITLE
OCPBUGS-18091: [release-4.13] Update ignition parsing to prevent crash on cluster upgrade

### PR DESCRIPTION
**Pull request by sebsoto**
Contains various backports to ensure that WMCO 8 is compatible with the ignition specs present in both OCP 4.13 and 4.14.

### CODE REVIEW

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

Thanks for the quick fix

File: pkg/ignition/ignition.go [hunk 1]
```diff
@@ -6,8 +6,8 @@ import (
 	"sort"
 	"strings"
 
-	ignCfg "github.com/coreos/ignition/v2/config/v3_2"
-	ignCfgTypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	ignCfg "github.com/coreos/ignition/v2/config/v3_4"
+	ignCfgTypes "github.com/coreos/ignition/v2/config/v3_4/types"
```
Comment (jrvaldes) [inline pkg/ignition/ignition.go:10]: I don't think this is needed. [xref](https://github.com/openshift/windows-machine-config-operator/pull/1728#issue-1850346835)

cc @jkyros   Thoughts?

Comment (jkyros) [reply pkg/ignition/ignition.go:10]: Assuming upgrade order is such that the WMCO gets upgraded _after_ the MCO, there is potential at least for the 4.13 WMCO to receive a `3.4.0` version from the 4.14 MCO during an upgrade, so having the 4.13 MCO understand `3.4.0` seems like a good idea?

**Pull request merged**